src/Microsoft.AspNetCore.Routing/DefaultLinkGenerator.cs

```diff
@@ -169,24 +180,24 @@ internal sealed class DefaultLinkGenerator : LinkGenerator
 
             return GetUriByEndpoints(
                 endpoints,
-                ambientValues: null,
                 values,
-                scheme,
-                host,
-                pathBase,
-                fragment,
-                options);
+                ambientValues: null,
+                scheme: scheme,
+                host: host,
+                pathBase: pathBase,
+                fragment: fragment,
+                options: options);
         }
 
-        public override LinkGenerationTemplate GetTemplateByAddress<TAddress>(TAddress address)
+        public override LinkGenerationTemplate GetTemplateByAddress<TAddress>(TAddress address, LinkGenerationTemplateOptions options = default)
```

Comment:
What are the rules on default vs null? Elseware we use null, e.g. LinkOptions options = null.

Reply:
default means null for reference types. I just make this code self-consistent.